src/Validation/Validation.php

```diff
@@ -1082,6 +1082,76 @@ public static function uploadedFile($file, array $options = [])
     }
 
     /**
+     * Validates the size of an uploaded image.
+     *
+     * @param array $file The uploaded file data from PHP.
+     * @param array $options Options to validate width and height.
+     * @return bool
+     */
+    public static function imageSize($file, $options)
+    {
+        if (!isset($options['height']) && !isset($options['width'])) {
+            throw new InvalidArgumentException('Invalid image size validation parameters! Missing `width` and / or `height`!');
+        }
+
+        list($width, $height) = getimagesize($file['tmp_name']);
```

Comment:
I'd think that in a regular request CakePHP should never see such objects, but I guess that handling them generally wouldn't hurt in case CakePHP is being advertised with PSR7 support.